src/Git.jl

```diff
@@ -63,9 +63,9 @@ function iscommit(sha1s::Vector; dir="")
 end
 
 immutable State
-    head::String
-    index::String
-    work::String
+    head::AbstractString
```

Comment:
Using an abstract type for a field is generally not a good idea for performance. Better use `Compat.ASCIIString`.